Userland/Libraries/LibGfx/ImageFormats/TinyVGLoader.cpp

```diff
@@ -0,0 +1,544 @@
+/*
+ * Copyright (c) 2023, MacDue <macdue@dueutil.tech>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Array.h>
+#include <AK/Endian.h>
+#include <AK/FixedArray.h>
+#include <AK/LEB128.h>
+#include <AK/MemoryStream.h>
+#include <AK/Variant.h>
+#include <LibCore/File.h>
+#include <LibGfx/AntiAliasingPainter.h>
+#include <LibGfx/ImageFormats/TinyVGLoader.h>
+#include <LibGfx/Line.h>
+#include <LibGfx/Painter.h>
+#include <LibGfx/Point.h>
+
+namespace Gfx {
+
+using VarUInt = LEB128<u32>;
+
+static constexpr Array<u8, 2> TVG_MAGIC { 0x72, 0x56 };
+
+enum class ColorEncoding : u8 {
+    RGBA8888 = 0,
+    RGB565 = 1,
+    RGBAF32 = 2,
+    Custom = 3
+};
+
+enum class CoordinateRange : u8 {
+    Default = 0,
+    Reduced = 1,
+    Enhanced = 2
+};
+
+enum class StyleType : u8 {
+    FlatColored = 0,
+    LinearGradient = 1,
+    RadialGradinet = 2
+};
+
+enum class Command : u8 {
+    EndOfDocument = 0,
+    FillPolygon = 1,
+    FillRectangles = 2,
+    FillPath = 3,
+    DrawLines = 4,
+    DrawLineLoop = 5,
+    DrawLineStrip = 6,
+    DrawLinePath = 7,
+    OutlineFillPolygon = 8,
+    OutlineFillRectangles = 9,
+    OutLineFillPath = 10
+};
+
+struct FillCommandHeader {
+    u32 count;
+    TinyVGDecodedImageData::Style style;
+};
+
+struct DrawCommandHeader {
+    u32 count;
+    TinyVGDecodedImageData::Style line_style;
+    float line_width;
+};
+
+struct OutlineFillCommandHeader {
+    u32 count;
+    TinyVGDecodedImageData::Style fill_style;
+    TinyVGDecodedImageData::Style line_style;
+    float line_width;
+};
+
+enum class PathCommand : u8 {
+    Line = 0,
+    HorizontalLine = 1,
+    VerticalLine = 2,
+    CubicBezier = 3,
+    ArcCircle = 4,
+    ArcEllipse = 5,
+    ClosePath = 6,
+    QuadraticBezier = 7
+};
+
+struct TinyVGHeader {
+    u8 version;
+    u8 scale;
+    ColorEncoding color_encoding;
+    CoordinateRange coordinate_range;
+    u32 width;
+    u32 height;
+    u32 color_count;
+};
+
+static ErrorOr<TinyVGHeader> decode_tinyvg_header(Stream& stream)
+{
+    Array<u8, 2> magic_bytes;
```

Comment:
very nit:
Why not an u16 instead of an Array?

Reply:
Mainly because the spec defines it like (`[2]u8`)